src/parallel_simulation.rs

```diff
@@ -74,6 +81,19 @@ impl Simulation {
             self.scenario.network.neighbors(),
         );
 
+        let mut router: Option<Router> = None;
```

Comment:
It is possible to do it this way. Though all the other stuff which doesn't change over the course of the simulation like `network`, `population` and so on are members of the `Simulation` struct. I would think that router could be a member as well. Then, the router could be initialized outside of `Simulation` and be passed through the constructor of `Simulation`. 

Seperating the two concerns of constructing dependencies and running the simulation loop.

Reply:
I created a #32 for that because it's not that simple unfortunately.